src/components/auth/AuthContext.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { createContext, useContext, useState, type ReactNode } from 'react';
+import { getFromLocalStorage } from '../../utils/storageUtils';
+import type { User } from '../../types/User/types';
+
+interface AuthContextType {
+  user: User | null;
+  setUser: React.Dispatch<React.SetStateAction<User | null>>;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
+  const [user, setUser] = useState<User | null>(() => {
+    return getFromLocalStorage<User>('currentUser');
+  });
```

Comment:
This is not the right way to define the state
```suggestion
const userData = getFromLocalStorage<User>('currentUser');
  const [user, setUser] = useState<User | null>(userData);
```

Reply:
done . this is my commit id b6404a2ee781d8bb1b5122d05a141e4974c060a3